File: RouteZoekerC/Stad.cs
using System.Collections.Generic;
using System.Drawing;

class Stad : IComparer<Weg>
{
    public string Naam;
    public ICollection<Weg> Wegen;
    public Point Plek;
    private static Font font = new Font("Tahoma", 10);

    public Stad(string naam, Point plek)
    {
        Naam = naam;
        Plek = plek;
        Wegen = new LinkedList<Weg>();
    }
    public void BouwWeg(Stad doel, int kosten)
    {
        Wegen.Add(new Weg(doel, kosten));
    }
    public void Teken(Graphics gr, Brush brush)
    {
        gr.FillRectangle(brush, new Rectangle(Plek-new Size(5,5), new Size(10,10)));
        gr.DrawString(Naam, font, brush, Plek+new Size(6,-15));
    }

    // Implementatie van IComparer:
    // orden twee wegen in *dalende* volgorde
    // wat betreft de hemelsbrede afstand van hun doel tot deze Stad

    public int Compare(Weg a, Weg b)
    {
        return afstand(b.Doel) - afstand(a.Doel);
    }
    int afstand(Stad a)
    {
        return Stad.afstand(a.Plek, Plek);
    }
    static int afstand(Point a, Point b)
    {
        return kwadraat(a.X - b.X) + kwadraat(a.Y + b.Y);
        // Pythagoras zegt dat we ook nog de wortel moeten trekken,
        // maar dat doen we lekker niet, omdat het alleen om de ordening
        // van de afstanden gaat. Die verandert niet door het worteltrekken.
    }
    static int kwadraat(int x)
    {
        return x * x;
    }
}

File: RouteZoekerP/Stad.py
from PIL.ImageDraw import Draw
from PIL.ImageFont import truetype
from PIL.ImageTk   import PhotoImage
from PIL           import Image
from Weg import Weg

font = truetype("tahoma.ttf", size=10)

def dist(p1, p2):  # standaardfunctie in math vanaf Python 6.8
    (x1,y1) = p1
    (x2,y2) = p2
    return (x1-x2)**2 + (y1-y2)**2
    # Pythagoras zegt dat we ook nog de wortel moeten trekken
    # maar dat doen we lekker niet, omdt het alleen om de ordening
    # van de afstanden gaat. Die verandert niet door het worteltrekken.


class Stad:
    def __init__(self, naam, plek):
        self.Naam = naam
        self.Plek = plek
        self.Wegen = list()

    def BouwWeg(self, doel, kosten):
        self.Wegen.append(Weg(doel, kosten))

    def Teken(self, draw, kleur):
        (x, y) = self.Plek
        draw.rectangle(((x-5,y-5),(x+5,y+5)), kleur)
        draw.text((x+6, y-15), self.Naam, fill=kleur, font=font)

    def AfstandWegdoelTotStad(self, pad):
        return dist(pad.Doel.Plek, self.Plek)
